src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/collection-edit-overlay/main.js

```diff
@@ -0,0 +1,180 @@
+/*
+ * This file is part of the Sulu CMF.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define([
+    'services/sulumedia/collection-manager',
+    'services/sulumedia/overlay-manager'], function(CollectionManager, OverlayManager) {
+
+    'use strict';
+
+    var namespace = 'sulu.collection-edit.',
+
+        defaults = {
+            instanceName: '',
+            locale: app.sandbox.sulu.user.locale
+        },
+
+        constants = {
+            editFormSelector: '#collection-settings'
+        },
+
+        /**
+         * raised when the overlay get closed
+         * @event sulu.collection-edit.closed
+         */
+        CLOSED = function() {
+            return createEventName.call(this, 'closed');
+        },
+
+        /**
+         * raised when component is initialized
+         * @event sulu.collection-edit.closed
+         */
+        INITIALIZED = function() {
+            return createEventName.call(this, 'initialized');
+        },
+
+        /** returns normalized event names */
+        createEventName = function(postFix) {
+            return namespace + (this.options.instanceName ? this.options.instanceName + '.' : '') + postFix;
+        };
+
+    return {
+
+        templates: [
+            '/admin/media/template/collection/settings'
+        ],
+
+        /**
+         * Initializes the overlay component
+         */
+        initialize: function() {
+            // extend defaults with options
+            this.options = this.sandbox.util.extend(true, {}, defaults, this.options);
+
+            if (!this.options.collectionId) {
```

Comment:
@nnatter is this needed. Have you thought about the possibliity to throw an error